DragonSlayer.h

```diff
@@ -1,8 +1,14 @@
 #pragma once
-
+#include "Character.h"
+#include "AttackItem.h"
 struct DragonSlayer : Character
 {
+    DragonSlayer(const std::string&, int, int);
     void attack(Character& other) override;
+    const std::string& getName() override;
+    std::string getStats() override;
+    AttackItem attackItem;
```

Comment:
Your `attackItem` needs to be wrapped in a `unique_ptr` so it can be `reset()` after use.
- [ ] revise